### DISCUSSION TITLE
Integrating an arbitrary API that deals with model selection and conversation handling

### DISCUSSION THREAD

**Discussion by lrtherond** [danny-avila/LibreChat#3013] [Feature Requests & Suggestions]
The subject gave the body away almost.

I need a front-end, but I don't need the front-end to deal with model selection, conversation handling, etc.

I need the front-end to integrate with an arbitrary API using JWT-based auth.

All the front-end needs to know is: How to make a POST request to generate the next LLM turn.

Is this possible with LibreChat?

**Comment by danny-avila** [comment]
The frontend keeps track of many facets of the conversation state, not just the AI interactions, like previous conversations, available models, messages, presets, users, formatting of messages specific to the project, etc. All of this is very tightly coupled to the backend.

You can use any OpenAI-like API with this project, and could likely achieve exactly what you want, but the frontend doesn't do this alone. Most of the AI API interaction is handled by the backend, which tracks the turn of messages and what conversations they're tied to, and the relationship between messages (parent/child).

You can see the many different examples that are known to work with LibreChat here:
https://www.librechat.ai/docs/configuration/librechat_yaml/ai_endpoints

Not sure if this completely answers your question. To summarize, you can think of the entire project working much like ChatGPT but with the ability to connect virtually every AI provider out there, especially OpenAI-like, that is, API that conforms to the OpenAI API specs.

**Reply by lrtherond** [reply]
Thank you. I am currently reading the code starting with: BaseClient.js

Maybe, I can make something like PluginsClient.js that would make the API conform to what is expected.
